### PR TITLE
Request form - retrieve items for resource sharing

### LINKED ISSUE #3002: Provide option to request an unavailable item in temporary location `RES_SHARE$IN_RS_REQ`
https://bibdata.princeton.edu/bibliographic/99118399983506421/holdings/22555936970006421/availability.json

https://catalog.princeton.edu/catalog/99118399983506421

Request form should give the option to request from resource sharing.

### PR DESCRIPTION
**Pull request by christinach**
closes #3002 
In the request form, retrieve items for resource sharing location, `RES_SHARE$IN_RS_REQ`

### CODE REVIEW

**Comment by christinach** [comment]
see https://catalog-qa.princeton.edu/requests/9991807103506421?mfhd=22696270550006421

Screen Shot 2022-06-02 at 8 46 39 PM

This is in https://catalog-qa.princeton.edu/ which retrieves availability from alma production. I made sure to index the record against catalog-qa so that we can see the example record. 
I haven't updated the status in JS yet but it can be done in a separate PR. 
This PR allows patrons to request the item using the request form.

**Comment by christinach** [comment]
Do we want the request form to display 'Firestone Stacks' or is it ok to display 'Resource Sharing'. The record page displays 'Firestone Stacks'.

**Review by maxkadel — CHANGES_REQUESTED** [review]
Can we add a test for this behavior? I'm still a little confused about what the expected behavior for these items is, and that might help clarify them.

**Comment by mzelesky** [comment]
I would think we'd want the request form to display the same thing as the record page.

**Comment by christinach** [comment]
> 

@maxkadel I didnt open the PR for review because there was no test. I have added one now and will open it for review.

**Comment by christinach** [comment]
> I would think we'd want the request form to display the same thing as the record page.

Will do @mzelesky thanks for checking the screenshot

**Comment by christinach** [comment]
I've update the location display in the request form per @mzelesky request:

Screen Shot 2022-06-03 at 11 22 57 AM

**Comment by christinach** [comment]
This PR will allow patrons to request items from resource sharing. 
@carolyncole @maxkadel a heads up that availability is failing in this record which means: no status in the record page and the location we are seeing (Firestone Library - Stacks ) even though it's the desired one, it displays because it's the fall back look up to [Bibdata](https://bibdata.princeton.edu/locations/holding_locations.json) when the Js availability response is failing. 

I will work on a separate PR to update availability status and make sure that the location display is the permanent location.

**Review by maxkadel — APPROVED** [review]

File: app/models/requests/request.rb [hunk 1]
```diff
@@ -290,7 +290,7 @@ def build_requestable_mfhd_item(_requestable_items, holding_id, item, barcodesor
         item_loc = item_current_location(item)
         current_location = get_current_location(item_loc: item_loc)
         item['status_label'] = barcodesort[item['barcode']][:status_label] unless barcodesort.empty?
-        calculate_holding = if item["in_temp_library"]
+        calculate_holding = if item["in_temp_library"] && item["temp_location_code"] != "RES_SHARE$IN_RS_REQ"
```
Comment (carolyncole) [inline app/models/requests/request.rb:293]: Can this utilize the helper method? `.in_resource_sharing?`

**Review by carolyncole — CHANGES_REQUESTED** [review]
@christinach Just one question

**Review by carolyncole — APPROVED** [review]

**Pull request merged**